server/src/main/java/io/crate/analyze/TableElementsAnalyzer.java

```diff
@@ -648,6 +617,38 @@ private void processConstraint(RefBuilder builder, ColumnConstraint<Expression>
                     throw new IllegalArgumentException(String.format(Locale.ENGLISH,
                         "Column \"%s\" is declared as NOT NULL, therefore, cannot be declared NULL", columnName));
                 }
+            } else if (constraint instanceof DefaultConstraint<Expression> defaultConstraint) {
```

Comment:
When I refactored the TableElementsAnalyzer I intentionally moved away from using visitors for everything here because it made it actually more complicated due to state access limitations

Do you see a concrete advantage of introducing one here again? 
If not, maybe instead we can explore using sealed classes for the constraints and use switch pattern matching to get exhaustiveness compiler checks

Reply:
I see, thx for the info. Honestly, I don't think it has an actual benefit.
Is there any benefit (performance wise) at runtime with the sealed classes vs the plain instanceof?

Reply:
performance is afaik roughly the same. The main benefit is exhaustiveness check. No default clause + compiler failure as soon as a new permited class is introduced.

Reply:
As `ColumnConstraint` is extending from `Node`, introducing sealed classed would have to be done on the `Node`. This would require a quite huge refactoring, which I'm not sure is worth the benefit. Or was this (refactoring almost all parser classes) the idea @mfussenegger?

Reply:
> As ColumnConstraint is extending from Node, introducing sealed classed would have to be done on the Node.

I thought you can use sealed on `ColumnConstraint` directly:

```
public abstract sealed class ColumnConstraint<T> extends Node
    permits
        CheckColumnConstraint,
        ColumnStorageDefinition,
        IndexColumnConstraint,
        NotNullColumnConstraint,
        NullColumnConstraint,
        PrimaryKeyColumnConstraint {
```

And then each of the concrete constraints need to be final.

But this shouldn't block this PR. I just meant that before going back to visitors, we _could_ explore that. Not that we should be doing it.

Reply:
> I thought you can use sealed on ColumnConstraint directly:

Yes you're right, can be easily done. I mixed up the syntax 🤦  ... pushed a fixup.